test/unit/configuration/configuration_parser_test.js

```diff
@@ -0,0 +1,139 @@
+    const fileContent = `{
+            "common": {
+                "log_level": "info",
+                "log_path": "/some-path/some-directory"
+            } 
+        }`;
+    await fsPromises.writeFile(filePath, fileContent, { encoding: 'utf8' });
+
+    // when
+    const configuration = await configurationParser.parse(filePath);
+
+    // then
+    assert.equal(configuration.logLevel, 'info');
+    assert.equal(configuration.logPath, '/some-path/some-directory');
+  });
+
+  it('should fail when wrong level', async function () {
+    // given
+    const fileName = 'config_wrong_level.json';
+    const filePath = path.join(tempDir, fileName);
+    const fileContent = `{
+            "common": {
+                "log_level": "unknown",
+                "log_path": "/some-path/some-directory"
+            } 
+        }`;
+    await fsPromises.writeFile(filePath, fileContent, { encoding: 'utf8' });
+
+    // expect
+    await assert.rejects(
+      async () => await configurationParser.parse(filePath),
+      (err) => {
+        assert.strictEqual(err.name, 'Error');
+        assert.strictEqual(err.message, 'Parsing easy logging configuration failed');
+        return true;
+      });
+  });
+
+  const noValueFileContents = [
+    `{
+            "common": {
+                "log_level": null,
+                "log_path": null
+            } 
+        }`,
+    `{
+            "common": {} 
+        }`
+  ];
+
+  noValueFileContents.forEach((fileContent, index) => {
+    it('should parse config without values ' + index.toString(), async function () {
+      // given
+      const fileName = 'config_nulls_' + index.toString() + '.json';
+      const filePath = path.join(tempDir, fileName);
+      await fsPromises.writeFile(filePath, fileContent, { encoding: 'utf8' });
+
+      // when
+      const configuration = await configurationParser.parse(filePath);
+
+      // then
+      assert.equal(configuration.logLevel, null);
+      assert.equal(configuration.logPath, null);
+    });
+  });
+
+  const notGivenFileNames = [null, '', undefined];
+
+  notGivenFileNames.forEach((filePath, index) => {
+    it('should return null when config file not given ' + index.toString(), async function () {
+      // when
+      const configuration = await configurationParser.parse(filePath);
+
+      // then
+      assert.strictEqual(configuration, null);
+    });
+  });
+
+  it('should fail when config file does not exist', async function () {
+    // expect
+    await assert.rejects(
+      async () => await configurationParser.parse('./not-existing-config.json'),
+      (err) => {
+        assert.strictEqual(err.name, 'Error');
+        assert.strictEqual(err.message, 'Finding easy logging configuration failed');
+        return true;
+      });
+  });
+
+  const wrongFileContents = [
+    `{
+            "common": {
+                "log_level": "unknown",
+                "log_path": "/some-path/some-directory"
+            } 
+        }`,
+    '{}'
```

Comment:
is it prohibited? should not we treat it as no config?

Reply:
in .NET I made it prohibited. The idea was that this config file has "common" configuration, and maybe also driver specific ones. So if there is no "common" section I treat such a config as invalid one.